packages/jest-config/src/__tests__/normalize.test.js

```diff
@@ -1477,9 +1477,27 @@ describe('moduleFileExtensions', () => {
   });
 });
 
+describe('cwd', () => {
+  it('is not lost if the config has its own cwd property', () => {
+    console.warn.mockImplementation(() => {});
+    const {options} = normalize(
+      {
+        rootDir: '/root/',
+        cwd: '/tmp/config-sets-cwd-itself',
+      },
+      {},
+    );
+    expect(options.cwd).toBe(process.cwd());
+    expect(console.warn).toHaveBeenCalled();
+  });
+});
+
 describe('Defaults', () => {
   it('should be accepted by normalize', () => {
-    normalize({...Defaults, rootDir: '/root'}, {});
+    // cwd is an exception, it must not appear in InitialOptions
+    const {cwd: _cwd, ...defaults} = Defaults;
+
+    normalize({...defaults, rootDir: '/root'}, {});
```

Comment:
I don't follow - shouldn't options just override default options?

Reply:
As I understand it, `cwd` is not a supported option for users to specify in the config. I mentioned in the issue that it does not appear in the docs or in `InitialOptions`

Reply:
@rubennorte you added it in #7146 - thoughts?

Reply:
(I also think it's right that it is not something that can be explicitly set, that should only be the case for `rootDir`)

Reply:
Either way, `cwd` becoming `undefined` if it is explicitly set is definitely not expected behavior 😅